system-tests/lib/worker/background_worker.go

```diff
@@ -0,0 +1,164 @@
+package worker
+
+import (
+	"fmt"
+	"os"
+	"runtime/debug"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+
+	libformat "github.com/smartcontractkit/chainlink/system-tests/lib/format"
+)
+
+// Future represents a typed result from a background task
+type Future[T any] struct {
+	value   T
+	err     error
+	done    chan struct{}
+	isReady bool
+	mutex   sync.RWMutex
+}
+
+// Get blocks until the task completes and returns the typed result
+func (f *Future[T]) Get() (T, error) {
+	<-f.done
+	f.mutex.RLock()
+	defer f.mutex.RUnlock()
+	return f.value, f.err
+}
+
+// IsReady returns true if the task has completed (non-blocking)
+func (f *Future[T]) IsReady() bool {
+	f.mutex.RLock()
+	defer f.mutex.RUnlock()
+	return f.isReady
+}
+
+// TypedBackgroundRunner manages execution of background tasks with full type safety
+type TypedBackgroundRunner struct {
```

Comment:
Nit: consider calling this BackgroundRunner :) It's not really relevant to the caller that it's typed.